ed25519-dalek/src/hazmat.rs

```diff
@@ -106,6 +120,18 @@ impl ExpandedSecretKey {
             .into()
         })
     }
+
+    /// Construct an `ExpandedSecretKey` from a scalar and hash prefix.
+    pub fn from_scalar_and_prefix(scalar: Scalar, hash_prefix: HashPrefix) -> Self {
```

Comment:
I thought `from_scalar_and_hash_prefix` was a bit unwieldy but it would be more consistent.

This method could probably use a longer comment explaining its use cases (e.g. Ed25519-BIP32).